EasyRestProjectNetTeam2/EasyRestTests/CheckAddEmploeeTest.cs

```diff
@@ -0,0 +1,70 @@
+        MenuPage menuPage;
+        ManageAdministratorPage manageAdministratorPage;
+        ManageWaitersPage manageWaitersPage;
+
+        [SetUp]
+        public override void SetUp()
+        {
+            base.SetUp();
+            homePage = GetHomePage();
+            homePage.HeaderMenuComponent.ClickSignInButton();
+            signInPage = GetSignInPage();
+            signInPage.SignInWithValidData(dataModel.EmailForOwner, dataModel.PasswordForOwner);
+            manageRestaurantsPage = GetManageRestaurantsPage();
+            manageRestaurantsPage.WaitAndClickMoreButton(dataModel.TimeToWait);
+            manageRestaurantsPage.WaitAndClickManageButton(dataModel.TimeToWait);
+            menuPage = GetMenuPage();
+        }
+
+        [Test]
+        public void CheckAddAdministrator()
+        {
+            menuPage.LeftBarComponent.WaitAndClickAdministratorsLeftBarButton(dataModel.TimeToWait);
+            manageAdministratorPage = GetManageAdministratorPage();
+            manageAdministratorPage.ClickPlusAdministrator();
+            manageAdministratorPage.AddEmploeeComponent.WaitAndSendKeysToInputData(dataModel.NameForNewEmploee,
+                dataModel.EmailForNewEmploee, dataModel.PasswordForNewEmploee,
+                dataModel.PhoneForNewEmploee, dataModel.TimeToWait);
+            manageAdministratorPage.AddEmploeeComponent.ClickAddNewEmploee();
+            var actualPageUrl = manageAdministratorPage.WaitAndGetTextFromAdministratorNameField(dataModel.TimeToWait);
+            var expectedSearchWord = dataModel.NameForNewEmploee;
+            StringAssert.Contains(expectedSearchWord, actualPageUrl, "Search word is absent in emploee name field");
+        }
+
+        [Test]
+        public void CheckAddWaiter()
+        {
+            menuPage.LeftBarComponent.WaitAndClickWaiterLeftBarButton(dataModel.TimeToWait);
+            manageWaitersPage = GetManageWaitersPage();
+            manageWaitersPage.ClickAddWaiterButton();
+            manageWaitersPage.AddEmploeeComponent.WaitAndSendKeysToInputData(dataModel.NameForNewEmploee,
+                dataModel.EmailForNewEmploee, dataModel.PasswordForNewEmploee,
+                dataModel.PhoneForNewEmploee, dataModel.TimeToWait);
+            manageWaitersPage.AddEmploeeComponent.ClickAddNewEmploee();
+            var actualPageUrl = manageWaitersPage.WaitAndGetTextFromWaitersNameField(dataModel.TimeToWait);
+            var expectedSearchWord = dataModel.NameForNewEmploee;
+            StringAssert.Contains(expectedSearchWord, actualPageUrl, "Search word is absent in emploee name field");
+        }
+
+        [TearDown]
+        public override void TearDown()
+        {
+            DatabaseManager.SendNonQuery(queryDataModel.SetPreviousAdministrator);
+            DatabaseManager.SendNonQuery(queryDataModel.DeleteUserByEmail, dataModel.EmailForNewEmploee);
+            DatabaseManager.SendNonQuery(queryDataModel.DeleteTokenByEmail, dataModel.EmailForOwner);
```

Comment:
are these methods exception safe? is it possible that `SendNonQuery()` might fail with exception and `driver.Close();` won't work?

Reply:
Yes, they are safe. There is one condition that DB must be reset and everybody in our team has done this so all condition is true.